src/ergw_aaa_radius.erl

```diff
@@ -870,7 +870,18 @@ send_request(Req, Session, #{server := NAS, retries := Retries, timeout := Timeo
 			{server_name, maps:get(server_name, Opts, Id)},
 			{retries, Retries},
 			{timeout, Timeout}],
-    eradius_client:send_request(NAS, Req, RadiusClientOpts).
+    case eradius_client:send_request(NAS, Req, RadiusClientOpts) of
+        {ok, _, _} = Res ->
+            Res;
+        _Err ->
+            %% otherwise NAS is map of server and/or server_backup entries
+            case eradius_client:send_request(maps:get(server_backup, Opts, NAS), Req, RadiusClientOpts) of
+                {ok, _, _} = Res ->
+                    Res;
+                _ ->
+                    {error, invalid_NAS, Opts}
+            end
+    end.
```

Comment:
That is not going to work for authentication.

There is nothing that says that the server and the backup server have to use the same shared secret.
The return from send_request is feed directly into radius_response and that method uses the shared secret from `server` to validate the answer.
Having send_request silently switching servers would break that validation if the shared secrets are different.

What is also problematic, is multi request challenge authentication as used by CAPWAP (which also uses ergw_aaa). For that all requests need to be sent to same server, switching server mid-exchange is not going to work.

Reply:
I see, thanks for explanation. If I got your point correct - fixed. Please re-review.